src/main/java/com/limechain/rpc/WsRpcClientImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.limechain.rpc;
+
+import com.limechain.constants.RpcConstants;
+import lombok.SneakyThrows;
+import org.teavm.jso.browser.Window;
+import org.teavm.jso.core.JSString;
+import org.teavm.jso.websocket.WebSocket;
+
+import java.util.ArrayDeque;
+
+/**
+ * The implementation of {@link WsRpcClient}. Uses a native JS Websocket implementation.
+ */
+public class WsRpcClientImpl implements WsRpcClient {
+
+    private WebSocket ws;
+    private final ArrayDeque<String> rpcResponses;
```

Comment:
Optional improvement: When using ArrayDeque, I prefer appending either ...Stack or ...Queue at the end of the name to clarify my intentions. Currently, I have to check the methods used to determine whether the Deque is functioning as a queue.